docs/spec/spec.md

```diff
@@ -267,6 +273,189 @@ The GraphQL engine acts as the main processing unit in the Ballerina GraphQL pac
+
+On retrieval of a successful response from the `execute` method, the client tries to perform data binding for the user-defined data type. The data type defined by the user should be a subtype of  `graphql:GenericResponseWithErrors|record{}|json`. Otherwise, the data binding fails with an error.
+
+The `GenericResponseWithErrors` is a predefined type in the graphql module that represents the shape of a graphql response.
+
+```ballerina
+# Represents the target type binding record with data, extensions and errors of a GraphQL 
+# response for `execute` method.
+#
+# + extensions -  Meta information on protocol extensions from the GraphQL server
+# + data -  The requested data from the GraphQL server
+# + errors - The errors occurred (if present) while processing the GraphQL request.
+
+public type GenericResponseWithErrors record {|
+   map<json?> extensions?;
+   record {| anydata...; |}|map<json?> data?;
+   ErrorDetail[] errors?;
+|};
+
+# Represents the details of an error that occurred during parsing, validating, or executing a GraphQL document.
+#
+# + message - The details of the error
+# + locations - The locations in the GraphQL document related to the error
+# + path - The GraphQL resource path of the error
+# + extensions - Additional information to errors
+public type ErrorDetail record {|
+    string message;
+    Location[] locations?;
+    (int|string)[] path?;
+    map<anydata> extensions?;
+|};
+```
+
+##### Example: Client Data Binding
+
+```ballerina
+import ballerina/graphql;
+
+// user defined types
+type ProfileResponseWithErrors record {|
+    *graphql:GenericResponseWithErrors;
+    ProfileResponse data;
+|};
+
+type ProfileResponse record {|
+    ProfileData one;
+|};
+
+type ProfileData record {
+    string name;
+};
+
+public function main() returns error? {
+    graphql:Client graphqlClient = check new ("http: //localhost:9091/graphql");
+    string document = "{ one: profile(id: 100) {name} }";
+    // data binding to user defined type
+    ProfileResponseWithErrors response = check graphqlClient->execute(document);
+    string name = response.data.one.name;
+}
+```
+
+#### 2.5.5 Client Error Handling
+
+Following are the error types specific to the `graphql:Client`, which can be returned during the client initialization or invocation of the execute method. 
+
+```ballerina
+# Represents GraphQL client related generic errors.
+public type ClientError distinct error;
+
+# Represents GraphQL client-side or network level errors.
+public type RequestError distinct ClientError;
+
+# Represents network-level errors.
+public type HttpError distinct (RequestError & error<record {| anydata body; |}>);
+
+# Represents GraphQL errors due to request validation.
+public type InvalidDocumentError distinct (RequestError & error<record {| ErrorDetail[]? errors; |}>);
+
+# Represents client-side data binding error.
+public type PayloadBindingError distinct (ClientError & error<record {| ErrorDetail[]? errors; |}>);
+```
```

Comment:
Let's not just copy and paste the Ballerina code here. Instead, use point form (as sub-topics) to describe each error type.

Reply:
Updated